config/foreman-proxy-content.migrations/230811192103-disable-katello-agent.rb

```diff
@@ -0,0 +1,3 @@
+if answers['foreman_proxy_content'].is_a?(Hash) && answers['foreman_proxy_content'].key?('enable_katello_agent')
+  answers['foreman_proxy_content'].delete('enable_katello_agent')
+end
```

Comment:
While this would remove it when the migration is ran the first time, if the parameter still exists coming from the puppet module, I am fairly confident the parameter would just come back. And users will see it as an option. The hiera config can work to ensure that no matter what the user sets it is ignored. 

I have been thinking we should drop the parameter (from puppet-katello and puppet-foreman_proxy_content) and hard-code the value inside the puppet module to false. In that way, we can ensure that the puppet module does all the right clean up but that the user does not see the installer option and cannot try to enable it.

Reply:
It will come back, but with the default value of `false`. I could remove the parameter from the modules however and hard-code the value to be false, if it's necessary to prevent users from attempting to re-enable it. (But Katello still won't let them use it if they tried)

Reply:
Flipping this: I don't see a reason to allow users to re-enable it, since Katello's code won't let you use it anyway.

Reply:
> Flipping this: I don't see a reason to allow users to re-enable it, since Katello's code won't let you use it anyway.

That was my thinking, if we can avoid even presenting the potential to users the better off we are. And I think we can do that in this scenario pretty easily.

Reply:
To prevent users from trying to re-enable it, this is roughly what the change looks like in puppet-katello: https://github.com/theforeman/puppet-katello/pull/471

(puppet-FPC in-transit)